### ISSUE TITLE
Player focus for Line of Sight'ed tokens

### ISSUE THREAD

**Issue by kandashi** [kandashi/Next-Up#5]

**Comment by jessev14** [comment]
Haven't had a chance to test this extensively, but I think you can just use `token.isVisible` to determine if the next combatant is visible from the player's owned token (I believe `token.isVisible` just returns true/false if the relevant token is/isn't visible on the canvas).

pseudocode:
```
if (nextCombatant.token.isVisible) {
    user.canvas.pan(nextCombatant.token.position)
} else {
    nothin'
}
```
I think pushing the canvas pan to each connected user will require some socket shenanigans which I don't have figured out, but there are examples out there, e.g. [Moerill's PnP](https://github.com/Moerill/fvtt-pointer).

**Comment by kandashi** [comment]
I hope I'd be able to just execute on the players side and ignore sockets

**Comment by jessev14** [comment]
woops, you can definitely just execute from the client side. Just need to adjust this bit (line 106 after my PR):

something like this should work:
```
if (!game.settings.get("Next-Up", "playerFocus")) {
    const firstGm = game.users.find((u) => u.isGM && u.active);
        if (firstGm && game.user === firstGm) {
```
where `playerFocus` is a new setting that is `false` by default.

This only issue left is seeing if the next combatant is actually in LOS or not. I'm not sure if my original idea of `token.isVisible` will work so easily, since you need the client's version of `canvas.tokens`, not the host/GM's version.

**Comment by kandashi** [comment]
My plan would be to separate the pan block to be non gm-only; then I think you could just use `character.getActiveTokens()` and compare sight with that?

**Comment by jessev14** [comment]
Separating the pan blocks seems like the better call. I think I've more or less solved it... standby for (another) PR.

**Comment by kandashi** [comment]
closed by #12
